torch/_inductor/bounds.py

```diff
@@ -87,9 +87,10 @@ def set_indirect(self, old, new):
     def get_index(self, name):
         expr = self.loop_body.indexing_exprs[name]
         bound = self.replacement_vals.get(expr)
-        if bound is not None:
-            return bound
-
-        bound = bound_sympy(expr, self.replacement_vals)
+        if bound is None:
+            bound = bound_sympy(expr, self.replacement_vals)
+        # The following assertion is true at the time of this writing
+        # We don't assert is as to not execute bound_sympy when bound is not None
+        # assert bound is None or bound == bound_sympy(expr, self.replacement_vals)
         self.replacement_vals[name] = bound
         return bound
```

Comment:
Couldn't most of this method just be reduced to this one liner using set_default https://www.w3schools.com/python/ref_dictionary_setdefault.asp?
```suggestion
        return self.replacement_vals.setdefault(name, bound_sympy(expr, self.replacement_vals))
```

Reply:
Actually, nvm that would always evaluate bound_sympy unfortunately which may be undesirable...

Reply:
That would eagerly execute `bound_sympy`, and that's exactly what we're trying to avoid.